### PR TITLE
Issue 3232 application reject with reason

### LINKED ISSUE #3232: BUILD - As a potential worker, I would like to know the reason why I was rejected for a bounty, so I can improve myself
### User Story

As a potential worker, I would like to know the reason why I was rejected for a bounty, so I can improve myself
### Why Is this Needed

[comment]: # (Describe the problem and why this task is needed. Provide description of the current state, what you would like to happen, and what actually happen)
*Summary*: I want to know why I was rejected

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

### Current Behavior
[comment]: # (Describe what actually happened.)
When a funder rejects a potential worker on a bounty, there is currently no information provided.

### Expected Behavior

[comment]: # (Describe what you expected to happen.)
When a funder rejects a potential worker on a bounty, a modal should pop up (#3231), and an email should be sent to the rejected worker describing the reason.

It should look something like this: 

work rejected email

- [ ] The email should be sent to the specific rejected worker.
- [ ] The email should show the issue logo and title.
- [ ] The email should provide one of the reasons below:

- You did not meet the desired qualifications.
- You did not provide sufficient information on the action plan.
- The bounty was available only under special circumstances (e.g. hackathon).
- The bounty was reserved for another person.
- The bounty was cancelled.
- (If 'Other' is selected on the modal, the free text provided by the funder should be displayed)

### Additional Information
[comment]: # (Any other information that would be useful, content, screenshots, etc.)
This build ticket is only for the **email** part. The modal itself is taken care of in #3231 .

Design ticket: #2773 
Sketch file: https://drive.google.com/open?id=14CpcEhilvDWlXG9lBuYSggaOe9zBNcdM

### PR DESCRIPTION
**Pull request by iamonuwa**
##### Refers/Fixes

Fixes #3232

### CODE REVIEW

File: app/retail/templates/emails/bounty_worker_rejected_with_reason.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{% load i18n  %}
+
+{% trans "The funder has denied your request to work on this issue.
+Checkout the issue explorer for other projects on the platform!
+ " %}
```
Comment (thelostone-mc) [inline app/retail/templates/emails/bounty_worker_rejected_with_reason.txt]: put them in the same line?

File: app/dashboard/views.py [hunk 2]
```diff
@@ -1594,13 +1596,13 @@ def bounty_details(request, ghuser='', ghrepo='', ghissue=0, stdbounties_id=None
...
-                helper_handle_admin_override_and_hide(request, bounty)
-                helper_handle_suspend_auto_approval(request, bounty)
-                helper_handle_mark_as_remarket_ready(request, bounty)
-                helper_handle_admin_contact_funder(request, bounty)
-                helper_handle_override_status(request, bounty)
+                # helper_handle_admin_override_and_hide(request, bounty)
+                # helper_handle_suspend_auto_approval(request, bounty)
+                # helper_handle_mark_as_remarket_ready(request, bounty)
+                # helper_handle_admin_contact_funder(request, bounty)
+                # helper_handle_override_status(request, bounty)
```
Comment (thelostone-mc) [inline app/dashboard/views.py]: Should these be commented out?

Comment (iamonuwa) [reply app/dashboard/views.py]: Damn! Forgot to uncomment after debugging. Updating it now

Comment (thelostone-mc) [inline app/dashboard/views.py]: looks like this wasn't updated ?

**Comment by iamonuwa** [comment]
@thelostone-mc fixed

File: app/assets/onepager/css/main.css [hunk 3]
```diff
@@ -560,6 +560,18 @@ body #main a:not(.button):hover {
   margin: auto;
 }
 
+.alert {
+  position: absolute;
+}
+#footer {
```
Comment (thelostone-mc) [inline app/assets/onepager/css/main.css]: add an empty line here ?

File: app/assets/onepager/css/main.css [hunk 4]
```diff
@@ -560,6 +560,18 @@ body #main a:not(.button):hover {
...
+  position: absolute;
+  background-color: #0D0764;
+  padding: 5px;
+  width: 100vw;
+  left: 0px;
+  text-align: center;
```
Comment (thelostone-mc) [inline app/assets/onepager/css/main.css]: use `text-center` class

File: app/assets/onepager/css/main.css [hunk 5]
```diff
@@ -560,6 +560,18 @@ body #main a:not(.button):hover {
   margin: auto;
 }
 
+.alert {
+  position: absolute;
+}
+#footer {
+  position: absolute;
```
Comment (thelostone-mc) [inline app/assets/onepager/css/main.css]: use `position-absolute` class

File: app/marketing/mails.py [hunk 6]
```diff
@@ -1023,6 +1024,23 @@ def start_work_approved(interest, bounty):
...
+    try:
+        setup_lang(to_email)
+        html, text, subject = render_bounty_worker_rejected_with_reason(interest, reason, bounty)
+
+        if not should_suppress_notification_email(to_email, 'bounty'):
+            send_mail(from_email, to_email, subject, text, html, categories=['transactional', func_name()])
+    finally:
+        translation.activate(cur_language)
 
 def start_work_rejected(interest, bounty):
```
Comment (thelostone-mc) [inline app/marketing/mails.py]: leave 2 empty lines as opposed to 1

File: app/marketing/mails.py [hunk 7]
```diff
@@ -1023,6 +1024,23 @@ def start_work_approved(interest, bounty):
     finally:
         translation.activate(cur_language)
 
+def bounty_worker_rejected_with_reason(interest, reason, bounty):
```
Comment (thelostone-mc) [inline app/marketing/mails.py]: leave 2 empty lines as opposed to 1

File: app/retail/emails.py [hunk 8]
```diff
@@ -1320,6 +1334,14 @@ def start_work_rejected(request):
...
     return HttpResponse(response_html)
 
+@staff_member_required
+def bounty_worker_rejected_with_reason(request):
+    from dashboard.models import Interest, Bounty
+    interest = Interest.objects.last()
+    bounty = Bounty.objects.last()
+    reason = request.POST.get('reason')
+    response_html, _, _ = render_bounty_worker_rejected_with_reason(interest, reason, bounty)
+    return HttpResponse(response_html)
```
Comment (thelostone-mc) [inline app/retail/emails.py]: leave 2 empty lines as opposed to 1

File: app/retail/emails.py [hunk 9]
```diff
@@ -1320,6 +1334,14 @@ def start_work_rejected(request):
     response_html, _, _ = render_start_work_rejected(interest, bounty)
     return HttpResponse(response_html)
 
+@staff_member_required
```
Comment (thelostone-mc) [inline app/retail/emails.py]: leave 2 empty lines as opposed to 1

File: app/retail/emails.py [hunk 10]
```diff
@@ -865,6 +865,20 @@ def render_start_work_approved(interest, bounty):
...
+        'interest': interest,
+        'bounty': bounty,
+        'reason': reason,
+        'approve_worker_url': bounty.approve_worker_url(interest.profile.handle),
+    }
+    subject = "Application Rejected"
+    response_html = premailer_transform(render_to_string("emails/bounty_worker_rejected_with_reason.html", params))
+    response_txt = render_to_string("emails/bounty_worker_rejected_with_reason.txt", params)
+
+    return response_html, response_txt, subject
```
Comment (thelostone-mc) [inline app/retail/emails.py]: leave 2 empty lines as opposed to 1

File: app/retail/emails.py [hunk 11]
```diff
@@ -865,6 +865,20 @@ def render_start_work_approved(interest, bounty):
 
     return response_html, response_txt, subject
 
+def render_bounty_worker_rejected_with_reason(interest, reason, bounty):
```
Comment (thelostone-mc) [inline app/retail/emails.py]: leave 2 empty lines as opposed to 1

**Comment by octavioamu** [comment]
cool, any video to see it working?

**Comment by iamonuwa** [comment]
> cool, any video to see it working?

Can't seem to get my video app running now. But everything works fine

**Comment by octavioamu** [comment]
@iamonuwa user https://chrome.google.com/webstore/detail/screen-and-webcam-recorde/jiihcciniecimeajcniapbngjjbonjan  is amazing.

**Comment by iamonuwa** [comment]
For some reasons, my PC has refused to allow me do a screen recording or screenshot @octavioamu

**Comment by danlipert** [comment]
@iamonuwa Can you make a recording with your phone? I'd really like to see that this works as expected.

**Comment by iamonuwa** [comment]
I'll do this on Monday

**Comment by thelostone-mc** [comment]
@iamonuwa ^ could you close this PR up and open a draft PR instead ?

**Comment by iamonuwa** [comment]
Alright

**Pull request closed without merging**